### ISSUE TITLE
How to run locally?

### ISSUE THREAD

**Issue by bergstar** [Sanster/IOPaint#469]
Is there a documentation on how to install / run after cloning the repo?

**Comment by Sanster** [comment]
In fact, you don't need to clone. Please check the readme first https://github.com/Sanster/IOPaint?tab=readme-ov-file#quick-start

**Comment by mhsin321** [comment]
you are very stupid u broke ourr heart poor people cant use itt ever in life bcz u put 5$ stpd price on it so sad i wanna run it localy or online both fail

**Comment by bergstar** [comment]
> In fact, you don't need to clone. Please check the readme first https://github.com/Sanster/IOPaint?tab=readme-ov-file#quick-start

Yes I did check it. I did also send you 5$ for the useless zip archive. I just need to know how to install and run it locally on windows. So I can "git pull" and do regular git stuff. ZIP archive dose not belong on Git.

If you need more money to make that happen, please let me know.

**Comment by wr4thful** [comment]
> > In fact, you don't need to clone. Please check the readme first https://github.com/Sanster/IOPaint?tab=readme-ov-file#quick-start
> 
> Yes I did check it. I did also send you 5$ for the useless zip archive. I just need to know how to install and run it locally on windows. So I can "git pull" and do regular git stuff. ZIP archive dose not belong on Git.
> 
> If you need more money to make that happen, please let me know.

both of you and @mhsin321 are braindead. 

open ur command prompt and type
`pip3 install torch==2.1.2 torchvision==0.16.2 --index-url https://download.pytorch.org/whl/cu118`
and then after it finishes type
`pip3 install iopaint`

and then to start it type the following
`iopaint start --model=lama --device=gpu --port=8080`

to make ur life easier open a notepad and copy the line above and save it as .bat file so whenever u want to run it u just need to run the bat file

thats it. the instruction are clear in the readme but you are either illiterate or just a moron.

**Comment by Sanster** [comment]
> > In fact, you don't need to clone. Please check the readme first https://github.com/Sanster/IOPaint?tab=readme-ov-file#quick-start
> 
> Yes I did check it. I did also send you 5$ for the useless zip archive. I just need to know how to install and run it locally on windows. So I can "git pull" and do regular git stuff. ZIP archive dose not belong on Git.
> 
> If you need more money to make that happen, please let me know.

I understand what you mean, you want to set up a local development environment, right? I have updated the README with some additions about the development environment. You can check it here: https://github.com/Sanster/IOPaint?tab=readme-ov-file#development.

BTW I really appreciate your support for my open-source project. If you think the installation package is of no use to you and you want a refund, please feel free to send an email to my mailbox(cwq1913@gmail.com) to let me know(the main purpose of sending the email is to find the order number through the email address in the payment backend.), and I will refund you immediately.

**Comment by bergstar** [comment]
> I understand what you mean, you want to set up a local development environment, right? I have updated the README with some additions about the development environment. You can check it here: https://github.com/Sanster/IOPaint?tab=readme-ov-file#development.

This is a great project and its your project, and I fully respect your descisissons on how you run and organize your project. Thanks for sharing details on how to run it in dev mode.

That said, from my point of view, any application which is somewhat complex (has a webserver) should not be handled as a package, since IOPaint is an application as it is intended to be used directly by a user.

Many users of IOPaint will probobly come from [AUTOMATIC1111](https://github.com/AUTOMATIC1111) or [comfyanonymous/ComfyUI](https://github.com/comfyanonymous/ComfyUI) - as many other projects like that. And based on that "heritage" what I would consider a normal workflow to get started would be:
```
git clone https://github.com/Sanster/IOPaint.git
pip install -r requirements.txt
python main.py 
```

And then documentation should also reflect basic usage, for instance for batch processing. How do I do that with 1-click install? or with dev install?

For what its worth my recommendation would be:
- Documented basic install and usage with git / pip / python 
- Paid 1-click conda for win / mac with added batch process tooling and "git update" scripts.

**Comment by wr4thful** [comment]
> > I understand what you mean, you want to set up a local development environment, right? I have updated the README with some additions about the development environment. You can check it here: https://github.com/Sanster/IOPaint?tab=readme-ov-file#development.
> 
> This is a great project and it's your project, and I fully respect your decisions on how you run and organize your project. Thanks for sharing details on how to run it in dev mode.
> 
> That said, from my point of view, any application which is somewhat complex (has a webserver) should not be handled as a package, since IOPaint is an application as it is intended to be used directly by a user.
> 
> Many users of IOPaint will probobly come from [AUTOMATIC1111](https://github.com/AUTOMATIC1111) or [comfyanonymous/ComfyUI](https://github.com/comfyanonymous/ComfyUI) - as many other projects like that. And based on that "heritage" what I would consider a normal workflow to get started would be:
> 
> ```
> git clone https://github.com/Sanster/IOPaint.git
> pip install -r requirements.txt
> python main.py 
> ```
> 
> And then documentation should also reflect basic usage, for instance for batch processing. How do I do that with 1-click install? or with dev install?
> 
> For what its worth my recommendation would be:
> 
> * Documented basic install and usage with git / pip / python
> * Paid 1-click conda for win / mac with added batch process tooling and "git update" scripts.
> 
> btw for the paid solution I would look into packaging it the same way as https://github.com/chaiNNer-org/chaiNNer

lmao is this a joke? or are u just illiterate? everything is well documented both in the repo and the website (https://www.iopaint.com). they even got video showcasing each feature and how to do it. theres instruction on how to do batch process. how to add more models or use more features. theres instruction for everything. and if u fail to understand something u can start an issue. it doesnt take a genius to understand these.

why do u even think about the necessity to clone the repo in the first place lol just go `pip3 install iopaint` and it takes care of everything. and if you've tried other UIs before u probably has installed most of the dependencies. which just makes the process even faster. the fact that u compared this project to other UIs is just braindead. im gonna say it, your perspective on how things should be just straight braindead and is less efficient than the current setup.

**Comment by mhsin321** [comment]
> > > In fact, you don't need to clone. Please check the readme first https://github.com/Sanster/IOPaint?tab=readme-ov-file#quick-start
> > 
> > 
> > Yes I did check it. I did also send you 5$ for the useless zip archive. I just need to know how to install and run it locally on windows. So I can "git pull" and do regular git stuff. ZIP archive dose not belong on Git.
> > If you need more money to make that happen, please let me know.
> 
> @mhsin321 no u stupid
> 
> open ur command prompt and type `pip3 install torch==2.1.2 torchvision==0.16.2 --index-url https://download.pytorch.org/whl/cu118` and then after it finishes type `pip3 install iopaint`
> 
> and then to start it type the following `iopaint start --model=lama --device=gpu --port=8080`
> 
> to make ur life easier open a notepad and copy the line above and save it as .bat file so whenever u want to run it u just need to run the bat file
> 
> thats it. the instruction are clear in the readme but you are either illiterate or just blind



> > > In fact, you don't need to clone. Please check the readme first https://github.com/Sanster/IOPaint?tab=readme-ov-file#quick-start
> > 
> > 
> > Yes I did check it. I did also send you 5$ for the useless zip archive. I just need to know how to install and run it locally on windows. So I can "git pull" and do regular git stuff. ZIP archive dose not belong on Git.
> > If you need more money to make that happen, please let me know.
> 
> @mhsin321 no u stupid
> 
> open ur command prompt and type `pip3 install torch==2.1.2 torchvision==0.16.2 --index-url https://download.pytorch.org/whl/cu118` and then after it finishes type `pip3 install iopaint`
> 
> and then to start it type the following `iopaint start --model=lama --device=gpu --port=8080`
> 
> to make ur life easier open a notepad and copy the line above and save it as .bat file so whenever u want to run it u just need to run the bat file
> 
> thats it. the instruction are clear in the readme but you are either illiterate or just blind


maybe you have a sick mind or an idiotic birth dot in you who cant make you in peace that why you misunderstood anyone
listen pal tnx for your unfortunate help but what  i saw was defiantly not a free gift that was fkn paid for 5$ which should not, it can be a donation rule but not pay also i try by myself google colab version but its showing errors fail to load ngok not work correctly also hugging face demo is shittt crap nothing option show even lama model is crap work there i download its standalone  version but its work like piece of rotten egg  models and plugins not load when i press select it load then show red error failed so in the end its need right development dude, get a life plz

**Comment by bergstar** [comment]
> why do u even think about the necessity to clone the repo in the first place lol just go `pip3 install iopaint` and it takes care of everything. 

Let the chatgpt answer to you:
The main difference between a Python package and an application lies in their purpose and structure:
**Package:**
Purpose: A package is intended to be reused. 
**Application**:
Purpose: An application is a complete, standalone program designed to perform a specific task or set of tasks. 

In summary, a package is a reusable collection of modules meant to be integrated into applications, while an application is a complete program designed for end-users, built using various packages and modules to provide specific functionalities.

**Comment by Sanster** [comment]
@bergstar I think it is quite normal to install an application through pip install, such as [tensorboard](https://github.com/tensorflow/tensorboard), [wandb](https://github.com/wandb/wandb), [Label Studio](https://labelstud.io/) and so on. IOPaint's frontend is not based on [gradio](https://www.gradio.app/), which means it not only depends on python, but also requires a nodejs environment to compile the frontend code in order to start debugging locally. The python package already includes the compiled frontend code, so users can easily use it directly. And when updating IOPaint, there is no need to execute `git pull`, just run `pip install -U iopaint` to update to a stable version (this is also how it is upgraded in the Windows installer). As I mentioned before, I have updated the process of starting the service locally after cloning the code in the README. If you encounter any issues, feel free to open an issue.
